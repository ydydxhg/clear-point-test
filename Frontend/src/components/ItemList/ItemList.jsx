import React from 'react'
import { Button, Table } from 'react-bootstrap'

const ItemList = (props) => {
  return (
    <>
      <h1>
        Showing {props.items ? props.items.length : 0} Item(s){' '}
        <Button variant="primary" className="pull-right" onClick={() => props.getItems()}>
          Refresh
        </Button>
      </h1>

      <Table striped bordered hover>
        <thead>
          <tr>
            <th>Id</th>
            <th>Description</th>
            <th>Action</th>
          </tr>
        </thead>
        <tbody>
          {props.items &&
            props.items.length > 0 &&
            props.items.map((item) => (
              <tr key={item.guid}>
                <td>{item.guid}</td>
                <td>{item.description}</td>
                <td>
                  <Button variant="warning" size="sm" onClick={() => props.handleMarkAsComplete(item)}>
                    Mark as completed
                  </Button>
                </td>
              </tr>
            ))}
        </tbody>
      </Table>
    </>
  )
}

export default ItemList
