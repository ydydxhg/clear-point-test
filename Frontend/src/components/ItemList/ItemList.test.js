import React from 'react'
import { cleanup, screen, fireEvent, render } from '@testing-library/react'
import ItemList from './ItemList'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it('should render without crashing', () => {
  render(<ItemList />)
})

it('should matches snapshot', () => {
  const tree = renderer
    .create(
      <ItemList
        items={[]}
        getItems={() => {
          return []
        }}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('should mark as complete', () => {
  var item = {
    guid: '0000',
    description: 'test',
    isCompleted: false,
  }
  let handleComplete = jest.fn()
  render(
    <ItemList
      items={[item]}
      getItems={() => {
        return [item]
      }}
      handleMarkAsComplete={handleComplete}
    />
  )
  var button = screen.getByText('Mark as completed')
  fireEvent.click(button)
  expect(handleComplete).toHaveBeenCalledTimes(1)
})
