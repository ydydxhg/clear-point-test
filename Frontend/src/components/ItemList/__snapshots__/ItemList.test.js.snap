// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches snapshot 1`] = `
Array [
  <h1>
    Showing 
    0
     Item(s)
     
    <button
      className="pull-right btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Refresh
    </button>
  </h1>,
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Id
        </th>
        <th>
          Description
        </th>
        <th>
          Action
        </th>
      </tr>
    </thead>
    <tbody />
  </table>,
]
`;
